frame/pablo/src/liquidity_bootstrapping_tests.rs

```diff
@@ -0,0 +1,672 @@
+) -> Validated<LiquidityBootstrappingPoolInfo<AccountId, AssetId, BlockNumber>, PoolIsValid<Test>> {
+	let pair = CurrencyPair::new(PROJECT_TOKEN, USDT);
+	let owner = ALICE;
+	let duration = MaxSaleDuration::get();
+	let start = 0;
+	let end = start + duration;
+	let initial_weight = MaxInitialWeight::get();
+	let final_weight = MinFinalWeight::get();
+	let fee = Permill::from_perthousand(1);
+	Validated::new(LiquidityBootstrappingPoolInfo {
+		owner,
+		pair,
+		sale: Sale { start, end, initial_weight, final_weight },
+		fee,
+	})
+	.expect("impossible; qed;")
+}
+
+fn with_pool<T>(
+	owner: AccountId,
+	sale_duration: BlockNumber,
+	initial_weight: Permill,
+	final_weight: Permill,
+	fee: Permill,
+	f: impl FnOnce(
+		PoolId,
+		&LiquidityBootstrappingPoolInfo<AccountId, AssetId, BlockNumber>,
+		&dyn Fn(BlockNumber),
+		&dyn Fn(),
+	) -> T,
+) -> T {
+	let random_start = 0xDEADC0DE;
+	let pair = CurrencyPair::new(PROJECT_TOKEN, USDT);
+	let end = random_start + sale_duration;
+	let pool = Validated::<_, PoolIsValid<Test>>::new(LiquidityBootstrappingPoolInfo {
+		owner,
+		pair,
+		sale: Sale { start: random_start, end, initial_weight, final_weight },
+		fee,
+	})
+	.expect("impossible; qed;");
+	new_test_ext().execute_with(|| -> T {
+		// Actually create the pool.
+		assert_ok!(Pablo::create(
+			Origin::signed(ALICE),
+			PoolInitConfiguration::LiquidityBootstrapping(pool.value())
+		));
+
+		// Will always start to 0.
+		let pool_id = 0;
+
+		// Relative to sale start.
+		let set_block = |x: BlockNumber| {
+			System::set_block_number(random_start + x);
+		};
+
+		// Forward to sale end.
+		let end_sale = || {
+			set_block(sale_duration + 1);
+		};
+
+		f(pool_id, &pool, &set_block, &end_sale)
+	})
+}
+
+fn within_sale_with_liquidity<T>(
+	owner: AccountId,
+	sale_duration: BlockNumber,
+	initial_weight: Permill,
+	final_weight: Permill,
+	fee: Permill,
+	initial_project_tokens: Balance,
+	initial_usdt: Balance,
+	f: impl FnOnce(
+		PoolId,
+		&LiquidityBootstrappingPoolInfo<AccountId, AssetId, BlockNumber>,
+		&dyn Fn(BlockNumber),
+		&dyn Fn(),
+	) -> T,
+) -> T {
+	with_pool(
+		owner,
+		sale_duration,
+		initial_weight,
+		final_weight,
+		fee,
+		|pool_id, pool, set_block, end_sale| -> T {
+			assert_ok!(Tokens::mint_into(PROJECT_TOKEN, &owner, initial_project_tokens));
+			assert_ok!(Tokens::mint_into(USDT, &owner, initial_usdt));
+
+			// Add initial liquidity.
+			assert_ok!(Pablo::add_liquidity(
+				Origin::signed(owner),
+				pool_id,
+				initial_project_tokens,
+				initial_usdt,
+				0_u128,
+				false
+			));
+
+			// Actually start the sale.
+			set_block(0);
+
+			f(pool_id, pool, set_block, end_sale)
+		},
+	)
+}
+
+mod create {
+	use super::*;
+
+	#[test]
+	fn arbitrary_user_can_create() {
+		new_test_ext().execute_with(|| {
+			assert_ok!(Pablo::create(
+				Origin::signed(ALICE),
+				PoolInitConfiguration::LiquidityBootstrapping(valid_pool().value())
+			),);
+		});
+	}
+
+	// #[test]
+	// fn admin_can_create() {
```

Comment:
Is this required? if yes then may be add some comments as TODO.

Reply:
It's just a test we need to enable after `cu-23v3155`. Let me add a comment